### PR TITLE
update houston ingress to support extra annotation

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

currently we are unable to set specific ingress annotation overrides for houston. This change allows users to add customer ingress annotation supported by ingress providers

## Related Issues

https://github.com/astronomer/issues/issues/5385

## Testing

Do not merge this PR until this text is replaced with details about how these changes were tested.

## Merging

0.31.2

### CODE REVIEW

**Comment by pgvishnuram** [comment]
need to add basic tests to validate ingress

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Code looks good. Needs test. PR description needs to be completed.

File: tests/chart_tests/test_houston_ingress.py [hunk 1]
```diff
@@ -72,3 +72,29 @@ def test_protect_houston_internal_urls(self, kube_version):
 }
 """
         )
+
+    def test_houston_ingress_overrides(self, kube_version):
+        custom_annotations = {
+            "nginx.ingress.kubernetes.io/upstream-keepalive-connections": "400",
+            "nginx.ingress.kubernetes.io/upstream-keepalive-timeout": "120",
```
Comment (danielhoherd) [inline tests/chart_tests/test_houston_ingress.py:79]: Let's change these to something conspicuous:
```suggestion
            "nginx.ingress.kubernetes.io/upstream-keepalive-connections": "999",
            "nginx.ingress.kubernetes.io/upstream-keepalive-timeout": "777",
```

**Review by danielhoherd — APPROVED** [review]
Shipit!

**Review by rishkarajgi — APPROVED** [review]

**Comment by rishkarajgi** [comment]
@pgvishnuram please cherry pick this to release branch

**Pull request merged**